packages/zoe/src/contractFacet/zcfSeat.js

```diff
@@ -146,27 +150,29 @@ export const createSeatManager = (zoeInstanceAdmin, getAssetKindByBrand) => {
       X`At least one seat has a staged allocation but was not included in the call to reallocate`,
     );
 
-    // No side effects above. All conditions checked which could have
-    // caused us to reject this reallocation.
-    // COMMIT POINT
-    // All the effects below must succeed "atomically". Scare quotes because
-    // the eventual send at the bottom is part of this "atomicity" even
-    // though its effects happen later. The send occurs in the order of
-    // updates from zcf to zoe, its effects must occur immediately in zoe
-    // on reception, and must not fail.
-    //
-    // Commit the staged allocations (currentAllocation is replaced
-    // for each of the seats) and inform Zoe of the
-    // newAllocation.
-
-    seats.forEach(commitStagedAllocation);
-
-    const seatHandleAllocations = seats.map(seat => {
-      const seatHandle = zcfSeatToSeatHandle.get(seat);
-      return { seatHandle, allocation: seat.getCurrentAllocation() };
+    zcfAssert.notThrows(() => {
+      // No side effects above. All conditions checked which could have
+      // caused us to reject this reallocation.
+      // COMMIT POINT
+      // All the effects below must succeed "atomically". Scare quotes because
+      // the eventual send at the bottom is part of this "atomicity" even
+      // though its effects happen later. The send occurs in the order of
+      // updates from zcf to zoe, its effects must occur immediately in zoe
+      // on reception, and must not fail.
+      //
+      // Commit the staged allocations (currentAllocation is replaced
+      // for each of the seats) and inform Zoe of the
+      // newAllocation.
+
+      seats.forEach(commitStagedAllocation);
+
+      const seatHandleAllocations = seats.map(seat => {
+        const seatHandle = zcfSeatToSeatHandle.get(seat);
+        return { seatHandle, allocation: seat.getCurrentAllocation() };
+      });
+
+      E(zoeInstanceAdmin).replaceAllocations(seatHandleAllocations);
     });
```

Comment:
Given the comment above, should this fail the vat later if the replaceAllocations breaks?  It makes me very uncomfortable to see a hanging async operation not get rolled up into "kill vat if this breaks"

Reply:
A failed `replaceAllocations` --- as well as several other failures that happen on the zoe side in service of a particular zcf vat --- must be fatal to that zcf vat. This PR is supposed to ensure that too. I'll leave this conversation unresolved until this specific case is dealt with.

Reply:
In startInstance.js, I did this for `replaceAllocations` specifically, so resolving. But @katelynsills @dtribble , What other zoe-side operations should, if they fail, kill their zcf?